### PR TITLE
Implement Cooperative Yielding Runtime Flag

### PR DESCRIPTION
**Pull request by adamgfraser**

### CODE REVIEW

File: core/shared/src/main/scala/zio/internal/FiberRuntime.scala [hunk 1]
```diff
@@ -279,7 +279,11 @@ class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs, runtim
           case trampoline: Trampoline =>
             trampolines = trampolines + 1
 
-            if (!trampoline.forceYield && (trampolines < FiberRuntime.MaxTrampolinesBeforeYield)) {
+            if (
```
Comment (jdegoes) [inline core/shared/src/main/scala/zio/internal/FiberRuntime.scala:282]: This is "trampoline yielding". There is another place inside `runLoop` where there is yielding: operation yielding. Both places have to be modified in order to completely disable cooperative yielding.

File: core/shared/src/main/scala/zio/RuntimeFlag.scala [hunk 2]
```diff
@@ -113,4 +113,14 @@ object RuntimeFlag {
     final val mask    = 1 << index
     final val notMask = ~mask
   }
+
+  /**
+   * The cooperative yielding flag determines whether the ZIO runtime will yield
+   * to another fiber.
+   */
+  case object CooperativeYielding extends RuntimeFlag {
```
Comment (jdegoes) [inline core/shared/src/main/scala/zio/RuntimeFlag.scala:121]: Looks good!

File: core/shared/src/main/scala/zio/internal/FiberRuntime.scala (hunk 1 above)
Comment (adamgfraser) [reply core/shared/src/main/scala/zio/internal/FiberRuntime.scala:282]: Doesn't that one just throw a `Trampoline` exception that we handle here?

Comment (jdegoes) [reply core/shared/src/main/scala/zio/internal/FiberRuntime.scala:282]: Ah yes, you are right, forgot I unified those two branches!

**Pull request merged**